### ISSUE TITLE
what result mean trainning is finished

### ISSUE THREAD

**Issue by skanger** [PatrickLib/captcha_recognition#18]
the loss is reduce to 0 when step 170 , but it don't stop, 
i pressed 'ctrl+c' breaked it, then try to eval and recognize,
but the result of recognize is so bad,

can i break the train when loss equal 0?
or it will finished , i need't do that?

**Comment by skanger** [comment]
now, it is step 2000 , every single step  loss is 0, but is keep traniing
my train set has only 3 pictures, but my mac is old, 
how can i  know it is finished

**Comment by LxKxC** [comment]
> now, it is step 2000 , every single step loss is 0, but is keep traniing
> my train set has only 3 pictures, but my mac is old,
> how can i know it is finished

* My captche program runs on raspberry PI 3B+.
* I use ```python captcha_gen_default.py``` to make captche，so I think it will stop in the 50000 step.
* It has been running for three days.
* You need to blow Rraspberry Pi with a small fan.If you don't, the Raspberry PI will burn down.

![train](https://user-images.githubusercontent.com/23060271/47624531-a6fcf200-db58-11e8-9674-4d365598386f.png)

**Comment by LxKxC** [comment]
> now, it is step 2000 , every single step loss is 0, but is keep traniing
> my train set has only 3 pictures, but my mac is old,
> how can i know it is finished

* I think if the training is complete, the training program will automatically save and exit, machine learning is a time-consuming process.

**Comment by skanger** [comment]
@LxKxC  I got  "Saving . in ./captcha_train/captcha" too.  Does it could be used as train set result?
if I keep trainning , it will  help the result better ?
Thanks for you reply.

**Comment by LxKxC** [comment]
@skanger I don't think that using the same picture to repeat training can improve the recognition rate.
Let me talk about my understanding:
* Picture A, training once.
* All features of the picture A have been counted in the program.
* Repeated training of the picture A, no new features found, so no help for training

**Comment by LxKxC** [comment]
@skanger I asked a friend from China who told me:
```随着训练次数的上升，残差平方和会下降，但你需要控制防止过拟合```
Translation into English generally means:
```As the number of training increases, the sum of squared residuals will decrease, but you need to control to prevent overfitting```

**Comment by tomorrow1pan** [comment]
Hello, in my PC, the train.py has runn over 20 hours, and step is over 19000 and the loss reduces to 0.00-0.01. Please can you tell me when the programe to stop？THX. @skanger @PatrickLib @LxKxC
